src/main/java/org/jboss/staxmapper/XMLMapper.java

```diff
@@ -40,7 +41,21 @@
      * @param name the element name
      * @param reader the reader which handles the element
      */
-    void registerRootElement(QName name, XMLElementReader<?> reader);
+    <T> void registerRootElement(QName name, XMLElementReader<T> reader);
```

Comment:
This should be fine, just do a javap to make sure the method is binary-compatible.